scripts/expX2.py

```diff
@@ -0,0 +1,763 @@
+                            (NRUNS_PER_SUT - completedForSut < 0.3 * state.perJob / sut.timeWeight)
+                         ):
+                        code += addJobBody(state.port, sut, seed, setting)
+                        state.updateBudget(sut.timeWeight)
+
+                    else:
+                        writeWithHeadAndFooter(code, state.port, sut, state.getTimeoutMinutes())
+                        state.resetTmpForNewRun()
+                        code = createOneJob(state, sut, seed, setting)
+
+                    # keep track that a new run has been handled
+                    completedForSut += 1
+
+        if state.opened:
+            writeWithHeadAndFooter(code, state.port, sut, state.getTimeoutMinutes())
+
+    print("Generated scripts: " + str(state.generated))
+    print("Max wait for a job: " + str(max(state.waits)) + " minutes")
+    print("Median wait for a job: " + str(statistics.median(state.waits)) + " minutes")
+    print("Budget left: " + str(state.budget))
+    print("Total time: " + str(sum(state.waits) / 60) + " hours")
+    print("Total budget: " + str(CPUS * sum(state.waits) / 60) + " hours")
+
+
+
+############################################################################
+### Custom
+### Following will need to be changed based on what kind of experiments
+### we want to run.
+############################################################################
+
+class ParameterSetting:
+    def __init__(self, name, values):
+        self.name = name
+        self.values = values
+        self.count = len(self.values)
+
+    def pvalue(self, index):
+        if index >= len(self.values):
+            exit("a value at the index "+ str(index) + " does not exist")
+        return (self.name, self.values[index])
+
+class Config:
+    def __init__(self, settings, filterKey=None):
+        self.filterKey = filterKey
+        self.settings = settings
+        self.numOfSettings = 1
+        for s in self.settings:
+            self.numOfSettings *= s.count
+
+    def genAllSettings(self):
+        all = []
+        lst = [0] * len(self.settings)
+        while lst is not None:
+            e = []
+            for i in range(len(lst)):
+                e.append(self.settings[i].pvalue(lst[i]))
+            all.append(e)
+            lst = self.plus1(lst)
+        return all
+
+
+    def plus1(self, lst):
+        if lst[0] < self.settings[0].count - 1:
+            lst[0] = lst[0] + 1
+            return lst
+        for i in range(len(lst)):
+            if lst[i] < self.settings[i].count-1:
+                lst[i] = lst[i] + 1
+                return lst
+            else:
+                lst[i] = 0
+        return None
+
+    def setting(self, index):
+        if index >= self.count:
+            exit("a setting at the index "+ str(index) + " does not exist")
+        return self.values[index]
+
+
+### NOTE there is no need to update this method
```

Comment:
in this case, remove this comment, and move the function before the `# Custom` comment that starts the are the user is supposed to modify manually